webui/src/main/java/org/polypheny/db/webui/Crud.java

```diff
@@ -3609,23 +3625,143 @@ void getTypeSchemas( final Context ctx ) {
     }
 
 
-    /**
-     * This method can be used to retrieve the status of a specific Docker instance and if
-     * it is running correctly when using the provided settings
-     */
-    public void testDockerInstance( final Context ctx ) {
-        String dockerIdAsString = ctx.pathParam( "dockerId" );
-        int dockerId = Integer.parseInt( dockerIdAsString );
+    void addDockerInstance( final Context ctx ) {
+        Map<String, Object> config = gson.fromJson( ctx.body(), Map.class );
+        DockerSetupResult res = DockerSetupHelper.newDockerInstance(
+                (String) config.getOrDefault( "host", "" ),
+                (String) config.getOrDefault( "alias", "" ),
+                (String) config.getOrDefault( "registry", "" ),
+                ((Double) config.getOrDefault( "communicationPort", (double) ConfigDocker.COMMUNICATION_PORT )).intValue(),
+                ((Double) config.getOrDefault( "handshakePort", (double) ConfigDocker.HANDSHAKE_PORT )).intValue(),
+                ((Double) config.getOrDefault( "proxyPort", (double) ConfigDocker.PROXY_PORT )).intValue(),
```

Comment:
Double looks so wrong here...

Reply:
When converting to Object, Gson converts JSON numbers to Double by default.